lib/middleware.js

```diff
@@ -597,7 +589,41 @@ module.exports = function middlewareConstructor() {
           return;
         }
 
-        req.session.publishUser = body;
+        req.user.publishId = body.id;
+        next();
+      });
+    },
+
+    /* Sets the project for the current request based on the project
+     * id provided as a parameter. The request's `user` property must
+     * be set by calling the `setUserIfTokenExists` middleware operation.
+     */
+    setProject: function(req, res, next) {
```

Comment:
I'm not sure that sound's great either! This function basically set's the current context. What I can do is the following:
```js
if(!user || !projectId) {
  req.project = null;
  next();
}
```